analysis/coverity-scan.sh

```diff
@@ -0,0 +1,124 @@
+#
+#     $ cd my-repo
+#     $ env COVERITY_SCAN_TOKEN="..." COVERITY_EMAIL="..." \
+#           GITHUB_ORG="..." GITHUB_PROJECT="..." \
+#           analysis/coverity-scan.sh
+#
+# Note: your Coverity project is assumed to be of the form:
+#
+#     "${GITHUB_ORG}/${GITHUB_PROJECT}"
+#
+# If that is not the case, you can override it as well on the command line via
+# the COVERITY_PROJECT env var.
+#
+# All other settings are either constants in this script or derived from the
+# repository, the `pom.xml` file, or other sources.
+#
+##############################################################################
+
+declare -r COVERITY_TOOL_TAR_GZ="coverity_tool.tar.gz"
+declare -r COVERITY_ANALYSIS_DIR="cov-analysis-linux64"
+
+# Required to be `cov-int` by Coverity docs:
+# https://scan.coverity.com/download
+declare -r COVERITY_BUILD_OUTPUT_DIR="cov-int"
+declare -r COVERITY_TAR_GZ="cov-int.tar.gz"
+
+if [ -z "${COVERITY_SCAN_TOKEN:-}" ]; then
+  echo "Error: env var COVERITY_SCAN_TOKEN not specified; exiting." >&2
+  exit 1
+elif [ -z "${COVERITY_EMAIL:-}" ]; then
+  echo "Error: env var COVERITY_EMAIL not specified; exiting." >&2
+  exit 2
+fi
+
+# Get the version number from the `pom.xml` file rather than hardcoding it here; see
+# https://stackoverflow.com/questions/41114695/get-pom-xml-version-with-xmllint
+# for more info. Note that simpler, traditional XPath-based solutions such as
+# https://stackoverflow.com/questions/15461737/how-to-execute-xpath-one-liners-from-shell
+# don't work, because Maven's `pom.xml` files use namespaces.
+declare -r PROJECT_VERSION="$(xmllint --xpath '/*[local-name()="project"]/*[local-name()="version"]/text()' pom.xml)"
+declare -r COVERITY_VERSION="${PROJECT_VERSION}/$(date +'%Y-%m-%dT%H:%M:%S')"
+
+declare -r GIT_HASH="$(git rev-parse HEAD)"
+declare -r COVERITY_DESCRIPTION="Automatic build upload via script; git hash: ${GIT_HASH}"
+
+declare -r GITHUB_ORG="${GITHUB_ORG:-JanusGraph}"
+declare -r GITHUB_PROJECT="${GITHUB_PROJECT:-janusgraph}"
+declare -r COVERITY_PROJECT="${GITHUB_ORG}%2F${GITHUB_PROJECT}"
+
+if ! [ -f "${COVERITY_TOOL_TAR_GZ}" ]; then
+  echo "Downloading Coverity analysis tool ..."
+  curl \
+    -X POST --data "token=${COVERITY_SCAN_TOKEN}&project=${COVERITY_PROJECT}" \
+    -o "${COVERITY_TOOL_TAR_GZ}" -s \
+    https://scan.coverity.com/download/linux64
+else
+  echo "Coverity tool tarball already exists; skipping download."
+fi
+
+if ! [ -d "${COVERITY_ANALYSIS_DIR}" ]; then
+  mkdir "${COVERITY_ANALYSIS_DIR}"
+  echo "Uncompressing archive ..."
+  # Sample file contents are of the form: "./cov-analysis-linux64-2017.07/bin/..."
+  tar zx --strip-components=2 --directory="${COVERITY_ANALYSIS_DIR}" -f "${COVERITY_TOOL_TAR_GZ}"
+else
+  echo "Coverity tool dir already exists; skipping uncompress step."
+fi
+
+if ! [ -d "${COVERITY_BUILD_OUTPUT_DIR}" ]; then
+  echo "Running Maven build with Coverity analysis ..."
+  mvn -DskipTests=true -T 12C -B -V clean
```

Comment:
I figured, why not clean faster? :-)

If you prefer `1C` I can change it to that, or perhaps scaled by number of CPUs.